pkg/render/fluentd_test.go

```diff
@@ -1051,3 +1083,106 @@ var _ = Describe("Tigera Secure Fluentd rendering tests", func() {
 		)
 	})
 })
+
+func setupEKSConfig() *render.EksCloudwatchLogConfig {
+	fetchInterval := int32(900)
+	return &render.EksCloudwatchLogConfig{
+		AwsId:         []byte("aws-id"),
+		AwsKey:        []byte("aws-key"),
+		AwsRegion:     "us-west-1",
+		GroupName:     "dummy-eks-cluster-cloudwatch-log-group",
+		FetchInterval: fetchInterval,
+	}
+}
+
+func setupInstallationConfig(t corev1.Toleration) *operatorv1.InstallationSpec {
+	return &operatorv1.InstallationSpec{
+		KubernetesProvider:      operatorv1.ProviderEKS,
+		ControlPlaneTolerations: []corev1.Toleration{t},
+	}
+}
+
+func getExpectedResourcesForEKS() []struct {
+	name    string
+	ns      string
+	group   string
+	version string
+	kind    string
+} {
+	return []struct {
+		name    string
+		ns      string
+		group   string
+		version string
+		kind    string
+	}{
+		{name: "tigera-fluentd", ns: "", group: "", version: "v1", kind: "Namespace"},
+		{name: render.FluentdPolicyName, ns: render.LogCollectorNamespace, group: "projectcalico.org", version: "v3", kind: "NetworkPolicy"},
+		{name: render.FluentdMetricsService, ns: render.LogCollectorNamespace, group: "", version: "v1", kind: "Service"},
+		{name: "eks-log-forwarder", ns: "", group: "rbac.authorization.k8s.io", version: "v1", kind: "ClusterRole"},
+		{name: "eks-log-forwarder", ns: "", group: "rbac.authorization.k8s.io", version: "v1", kind: "ClusterRoleBinding"},
+		{name: "eks-log-forwarder", ns: "", group: "policy", version: "v1beta1", kind: "PodSecurityPolicy"},
+		{name: "eks-log-forwarder", ns: "tigera-fluentd", group: "", version: "v1", kind: "ServiceAccount"},
+		{name: "tigera-eks-log-forwarder-secret", ns: "tigera-fluentd", group: "", version: "v1", kind: "Secret"},
+		{name: "eks-log-forwarder", ns: "tigera-fluentd", group: "apps", version: "v1", kind: "Deployment"},
+		{name: "tigera-fluentd", ns: "", group: "rbac.authorization.k8s.io", version: "v1", kind: "ClusterRole"},
+		{name: "tigera-fluentd", ns: "", group: "rbac.authorization.k8s.io", version: "v1", kind: "ClusterRoleBinding"},
+		{name: "tigera-fluentd", ns: "", group: "policy", version: "v1beta1", kind: "PodSecurityPolicy"},
+		{name: "fluentd-node", ns: "tigera-fluentd", group: "", version: "v1", kind: "ServiceAccount"},
+		{name: render.PacketCaptureAPIRole, ns: render.LogCollectorNamespace, group: "rbac.authorization.k8s.io", version: "v1", kind: "Role"},
+		{name: render.PacketCaptureAPIRoleBinding, ns: render.LogCollectorNamespace, group: "rbac.authorization.k8s.io", version: "v1", kind: "RoleBinding"},
+		// Daemonset
+		{name: "fluentd-node", ns: "tigera-fluentd", group: "apps", version: "v1", kind: "DaemonSet"},
+	}
+}
+
+func assertExpectedResourcesForEKS(deploy *appsv1.Deployment, t corev1.Toleration, cfg *render.FluentdConfiguration) {
+	Expect(deploy.Spec.Template.Spec.InitContainers).To(HaveLen(1))
```

Comment:
I'm not a fan of these types of helper functions - it makes it harder to debug the tests when they go wrong because line numbers now point here instead of the test itself.

You could use `ExpectWithOffset` instead, but then we lose the exact assertion that triggered it and instead only see the line in the calling test function.

I think we can leave this for now, but in the future I'd much rather we just have one or two mainline tests that verify all of these assertions, and then much smaller tests that assert on specific things.

There's no need for every multi-tenant env var test to check the SecurityContext, for example - we already have UTs that cover that. This is a good example: https://github.com/caseydavenport/operator/blob/3919c836a4518f6ff5569fcef9551f99930f3933/pkg/render/logstorage/linseed/linseed_test.go#L399-L423

^ That test only asserts the things that it is specifically looking for, keeping the test itself focused and simple.

Reply:
Done